examples/finetuning/dpo_tic_tac_toe/src/dpo_tic_tac_toe/dpo_workflow.py

```
@@ -0,0 +1,177 @@
+        Both players use TTC pipelines - the trained player uses an LLM-based
+        pipeline while the opponent uses random (or LLM) based pipeline.
+        All candidate moves from both players are recorded for DPO data.
+
+        Args:
+            role: "X" or "O" - which side the trained player plays
+
+        Returns:
+            Game outcome: "Win!", "Lose!", or "Draw!"
+        """
+        if role not in ["X", "O"]:
+            raise ValueError("Role must be either 'X' or 'O'.")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)